.github/workflows/cloud-dev-build-deploy.yml

```
@@ -0,0 +1,108 @@
+        run: |
+          docker push ${{ secrets.DOCKERHUB_USERNAME }}/aics-auth:${{ github.sha }}
+          docker push ${{ secrets.DOCKERHUB_USERNAME }}/aics-auth:latest
+
+  deploy-to-dev:
+    runs-on: ubuntu-latest
+    needs: build-docker-image
+    environment: dev
+
+    steps:
+      - name: checkout repository
+        uses: actions/checkout@v3
```

actionlint (1.7.9)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)